pkg/segment/query/processor/fillnullcommand_test.go

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2021-2024 SigScalr, Inc.
+//
+// This file is part of SigLens Observability Solution
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package processor
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/siglens/siglens/pkg/segment/query/iqr"
+	"github.com/siglens/siglens/pkg/segment/structs"
+	"github.com/siglens/siglens/pkg/segment/utils"
+	toputils "github.com/siglens/siglens/pkg/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+func insertColumnsWithSomeNulls(t *testing.T, iqr *iqr.IQR, fillValue utils.CValueEnclosure, columnsToInsert []string, valuesCountToInsert int) (
+	map[string][]utils.CValueEnclosure, map[string][]utils.CValueEnclosure) {
+	knownValuesToInsert := make(map[string][]utils.CValueEnclosure)
+	knownValuesToExpect := make(map[string][]utils.CValueEnclosure)
+
+	var err error
+
+	for _, column := range columnsToInsert {
+		valuesToInsert := make([]utils.CValueEnclosure, valuesCountToInsert)
+		valuesToExpect := make([]utils.CValueEnclosure, valuesCountToInsert)
```

Comment:
`knownValuesToInsert` and `knownValuesToExpect` look like they're always the same by the end of this function. Am I missing something? Also the `valuesToInsert` and `valuesToExpect` look the same

Reply:
This the code inside the loop

```
                         cValueToInsert := utils.CValueEnclosure{}
			var cValueToExpect utils.CValueEnclosure
			if insertNull {
				err = cValueToInsert.ConvertValue(nil)
				assert.NoError(t, err)
				cValueToExpect = fillValue
			} else {
				err = cValueToInsert.ConvertValue(fmt.Sprintf("value-%v", i))
				assert.NoError(t, err)
				cValueToExpect = cValueToInsert
			}

			valuesToInsert[i] = cValueToInsert
			valuesToExpect[i] = cValueToExpect
```

If `insertNull` is true, then `cValueToExpect` is assigned to the `fillValue` and `cValueToInsert` is set to `nil`. And if `insertNull` is same, then `cValueToExpect` and `cValueToInsert` are same.

When `fillNullProcessor.Process` is called, the `nil` will be filled by the `fillValue` and that is what I am expecting.

So No they would not end up same.